src/libethash/ethash.h

```diff
@@ -168,31 +150,30 @@ ethash_cache* ethash_light_get_cache(ethash_light_t light);
  *                 After this function concludes it will no longer have a cache.
  * @return         A pointer to the moved cache or NULL if there was no cache in the first place
  */
-ethash_cache* ethash_light_acquire_cache(ethash_light_t light);
+ethash_cache_t* ethash_light_acquire_cache(ethash_light_t light);
 
 /**
  * Allocate and initialize a new ethash_full handler
  *
- * @param dirname   The directory in which to put the DAG file.
- * @param seedhash  The seed hash of the block. Used in the DAG file naming.
- * @param params    The parameters to initialize it with. We are interested in
- *                  the full_size from here
- * @param cache     A cache object to use that was allocated with @ref ethash_cache_new().
- *                  Iff this function succeeds the ethash_full_t will take memory
- *                  ownership of the cache and free it at deletion. If not then the user
- *                  still has to handle freeing of the cache himself.
- * @param callback  A callback function with signature of @ref ethash_callback_t
- *                  It accepts an unsigned with which a progress of DAG calculation
- *                  can be displayed. If all goes well the callback should return 0.
- *                  If a non-zero value is returned then DAG generation will stop.
- * @return          Newly allocated ethash_full handler or NULL in case of
- *                  ERRNOMEM or invalid parameters used for @ref ethash_compute_full_data()
+ * @param dirname        The directory in which to put the DAG file.
+ * @param seedhash       The seed hash of the block. Used in the DAG file naming.
+ * @param full_size      The size of the full data in bytes.
+ * @param cache          A cache object to use that was allocated with @ref ethash_cache_new().
+ *                       Iff this function succeeds the ethash_full_t will take memory
+ *                       memory ownership of the cache and free it at deletion. If
+ *                       not then the user still has to handle freeing of the cache himself.
+ * @param callback       A callback function with signature of @ref ethash_callback_t
+ *                       It accepts an unsigned with which a progress of DAG calculation
+ *                       can be displayed. If all goes well the callback should return 0.
+ *                       If a non-zero value is returned then DAG generation will stop.
+ * @return               Newly allocated ethash_full handler or NULL in case of
+ *                       ERRNOMEM or invalid parameters used for @ref ethash_compute_full_data()
  */
 ethash_full_t ethash_full_new(
 	char const* dirname,
 	ethash_h256_t const* seed_hash,
-	ethash_params const* params,
-	ethash_cache const* cache,
+	uint64_t full_size,
```

Comment:
This would be possible but it would have 2 direct connotations:
1) Logic to derive full size from cache_size. Search in the data size tables perhaps?
2) Inability to provide testing sizes